### PR TITLE
Add 2023 summer capstone student data

### LINKED ISSUE #339: Add 23S COSC499 data

### PR DESCRIPTION
**Pull request by justino599**
Closes #339

### CODE REVIEW

File: benchmarking/data/real_data/cosc499_s2023_provider/providers.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    StudentProvider,
+    InitialTeamsProvider,
+)
+import json
+
+
+class COSC499S2023InitialTeamsProvider(InitialTeamsProvider):
+    def get(self) -> List[Team]:
+        """
+        Returns the teams that were created for the W2021T2 COSC341 class
```
Comment (Opeyem1a) [inline benchmarking/data/real_data/cosc499_s2023_provider/providers.py:18]: Wait like this returns the teams fr? The initial teams provider is for returning like shells of teams (I realized I gave the get() method the wrong return type)

Comment (Opeyem1a) [reply benchmarking/data/real_data/cosc499_s2023_provider/providers.py:18]: like for setting up the project requirements of each team shell and stuff

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: benchmarking/data/real_data/cosc499_s2023_provider/providers.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+from benchmarking.data.interfaces import (
+    StudentProvider,
+    TeamConfigurationProvider,
+)
+
+
+class COSC499S2023InitialTeamConfigurationProvider(TeamConfigurationProvider):
+    def get(self) -> TeamSet:
+        """
+        Returns the teams that were created for the W2021T2 COSC341 class
```
Comment (Opeyem1a) [inline benchmarking/data/real_data/cosc499_s2023_provider/providers.py:19]: nit; sus comment

File: benchmarking/data/real_data/cosc499_s2023_provider/providers.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        ) as f:
+            json_data = json.load(f)
+            teams = [TeamSerializer().decode(team) for team in json_data["teams"]]
+        return TeamSet(teams=teams)
+
+
+class COSC499S2023StudentProvider(StudentProvider):
+    def get(self, seed: int = None) -> List[Student]:
+        """
+        Returns a list of the students in COSC 341/51 in the W2021T2 semester.
```
Comment (Opeyem1a) [inline benchmarking/data/real_data/cosc499_s2023_provider/providers.py:32]: nit; sus comment

**Review by Opeyem1a — APPROVED** [review]
Merge if you need to, but some unupdated comments should be addressed eventually

File: benchmarking/data/real_data/cosc499_s2023_provider/providers.py (hunk 2 above)
Comment (justino599) [reply benchmarking/data/real_data/cosc499_s2023_provider/providers.py:19]: Not sure why this is sus unless you just want it to say TeamSet rather teams

Comment (Opeyem1a) [reply benchmarking/data/real_data/cosc499_s2023_provider/providers.py:19]: the class is wrong on both

Comment (Opeyem1a) [reply benchmarking/data/real_data/cosc499_s2023_provider/providers.py:19]: sorry should've been more specific in review haha

**Pull request merged**